### ISSUE TITLE
seed handmade NN architectures

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#150]
First figure out where and how to seed by hand...should we write it as a lisp? is that easy? Where does it plug in? Do we like how it plugs in? Can we make a script that goes from downloaded pretrained networks to extracted architectures saved in a way easily read by ezcgp?

Then make sure we have a solid and regular way to seed with the models we have in transfer learning.

Then start the other issue where we do research on types of architecture we may want to introduce into our nn. What networks are good for CNN image classification, other?

**Comment by roddtalebi** [comment]
Load in the architecture:
```
pretrained_model = tf.keras.applications.VGG16(include_top=False,
                                               weights='imagenet',
                                               input_tensor=None,
                                               pooling=None)
```

Translate to lisp:
```
def CNN_to_lisp(model):
    '''
    assumes tf.keras.applications Model
    mimics block_definition.get_lisp() especially for the args.
    and assumes model has no funky connections and is basically a linear connection
    '''
    lisp = "-1n"
    for i, layer in enumerate(model.layers[1:]):
        previous_lisp = deepcopy(lisp)

        # Conv2D
        if isinstance(layer, tf.keras.layers.Conv2D):
            operator = opTF.conv2D_layer
            arg_types = opTF.operator_dict[operator]["args"]
            args = []
            args.append(str(layer.filters))
            args.append(str(layer.kernel_size[0]))
            args.append(str(layer.activation))

        # MaxPool2D
        elif isinstance(layer, tf.keras.layers.MaxPool2D):
            operator = opTF.max_pool_layer
            arg_types = opTF.operator_dict[operator]["args"]
            args = []
            args.append(str(layer.pool_size[0]))
            args.append(str(layer.pool_size[1]))
            args.append(str(layer.strides[0]))

        else:
            print("not sure what this layer is...")
            pdb.set_trace()


        lisp = [operator.__name__, previous_lisp, *args]
    
    return lisp
```

**Comment by roddtalebi** [comment]
Got VGG16 working but need new primitives added to get the other NN to work

**Comment by roddtalebi** [comment]
where to store CNN_to_lisp()? move somewhere to a utilities script? change the name?

**Comment by roddtalebi** [comment]
@comrade813 
First, familiarize yourself with tensorflow ie Models class, layers class, conv2d method/class etc
Second, explore some other download-able pretrained networks https://www.tensorflow.org/api_docs/python/tf/keras/applications
Third, pick at random, or see which of the list are more commonly used, run it through the function we have above and see which tensorflow methods we are missing in our list of primitives
Fourth try to create those primitives

**Comment by roddtalebi** [comment]
@comrade813 use this branch
https://github.com/ezCGP/ezCGP/tree/feature/Issue150-seed_NN

**Comment by roddtalebi** [comment]
@comrade813 did some work on this branch https://github.com/ezCGP/ezCGP/tree/feature/Issue150-seed_NN
